batch/batch/batch.py

```diff
@@ -1029,22 +1029,23 @@ def is_successful(self):
     batch = await Batch.from_db(batch_id, user)
     if not batch:
         abort(404)
-    await batch.close()
+    asyncio.ensure_future(batch.close())
     return jsonify({})
 
+
 @prom_async_time(REQUEST_TIME_DELETE_BATCH)
 @routes.delete('/api/v1alpha/batches/{batch_id}')
 @rest_authenticated_users_only
 async def delete_batch(request, userdata):
     batch_id = int(request.match_info['batch_id'])
     user = userdata['username']
-
     batch = await Batch.from_db(batch_id, user)
     if not batch:
         abort(404)
-    await batch.mark_deleted()
+    asyncio.ensure_future(batch.mark_deleted())
```

Comment:
Is the caller not getting feedback on whether mark_deleted succeeded the design we want? Similar for the others. Are we going to give the caller a way to track whether the operations they requested succeeded or not (say subscribing to an event stream)

Reply:
I changed it so we've done all of the work that is not O(n) (updating the jobs), but still return success only when we've updated the batch state in the database.

Reply:
Yep, that sounds about right.

Reply:
Thanks!